### ISSUE TITLE
`useShorthandArrayType` doesn't align fully with eslint equivalent

### ISSUE THREAD

**Issue by jpike88** [rome/tools#4361]
### Environment information

```block
This rule is not behaving the same as the eslint equivalent.

Eslint rule below:

'@typescript-eslint/array-type': [
			'error',
			{
				default: 'array-simple',
			},
		],
```

See below screenshot from eslint typescript playground with the above rule enabled, correctly identifying the bottom example only. Rome would treat both examples as violating the rule.
<img width="566" alt="Screenshot 2023-04-08 at 11 32 07 am" src="https://user-images.githubusercontent.com/9585787/230703204-d12d06ef-8d50-412b-a649-db68598437fa.png">
```


### What happened?

see code example above

### Expected result

should be like eslint rule

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by unvalley** [comment]
If the only problem is false positives for that `object type`, then the issue is resolved by #4354

**Comment by jpike88** [comment]
ill close for now, give it spin when its out and go from there. cheers
